### PR TITLE
[New Resource] aws_identitystore_group_membership

### LINKED ISSUE #26939: [New Resource] aws_identitystore_group_membership
### Description

This new new resource is the next that will be required after identity_store_group [https://github.com/hashicorp/terraform-provider-aws/issues/26606](https://github.com/hashicorp/terraform-provider-aws/issues/26606) and identity_store_user [https://github.com/hashicorp/terraform-provider-aws/issues/26607](https://github.com/hashicorp/terraform-provider-aws/issues/26607) will be implemented.

That will make it possible to attach a user to a group.

### Requested Resource(s) and/or Data Source(s)

aws_identitystore_group_membership

### Potential Terraform Configuration

```terraform
resource "aws_identitystore_group_membership" "test" {
  identity_store_id = tolist(data.aws_ssoadmin_instances.test.identity_store_ids)[0]

  group_id  = aws_identitystore_group.test.group_id
  member_id = aws_identitystore_user.test.user_id
}
```

### References

https://github.com/hashicorp/terraform-provider-aws/issues/26598
https://github.com/hashicorp/terraform-provider-aws/issues/26606
https://github.com/hashicorp/terraform-provider-aws/issues/26607

### Would you like to implement a fix?

Yes

### PR DESCRIPTION
**Pull request by BenGisc**
Closes #26939.
Relates https://github.com/hashicorp/terraform-provider-aws/pull/26948.
Relates https://github.com/hashicorp/terraform-provider-aws/pull/26674.

Output from acceptance testing:

```
make testacc TESTS=TestAccIdentityStoreGroupMembership_ PKG=identitystore
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/identitystore/... -v -count 1 -parallel 20 -run='TestAccIdentityStoreGroupMembership_'  -timeout 180m
=== RUN   TestAccIdentityStoreGroupMembership_basic
=== PAUSE TestAccIdentityStoreGroupMembership_basic
=== RUN   TestAccIdentityStoreGroupMembership_disappears
=== PAUSE TestAccIdentityStoreGroupMembership_disappears
=== RUN   TestAccIdentityStoreGroupMembership_GroupId
=== PAUSE TestAccIdentityStoreGroupMembership_GroupId
=== RUN   TestAccIdentityStoreGroupMembership_MemberId
=== PAUSE TestAccIdentityStoreGroupMembership_MemberId
=== CONT  TestAccIdentityStoreGroupMembership_basic
=== CONT  TestAccIdentityStoreGroupMembership_MemberId
=== CONT  TestAccIdentityStoreGroupMembership_disappears
=== CONT  TestAccIdentityStoreGroupMembership_GroupId
--- PASS: TestAccIdentityStoreGroupMembership_disappears (28.90s)
--- PASS: TestAccIdentityStoreGroupMembership_basic (33.65s)
--- PASS: TestAccIdentityStoreGroupMembership_MemberId (51.80s)
--- PASS: TestAccIdentityStoreGroupMembership_GroupId (51.81s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/identitystore	56.587s

...
```

### CODE REVIEW

File: website/docs/r/identitystore_group_membership.html.markdown [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+In addition to all arguments above, the following attributes are exported:
+
+* `membership_id` - The identifier of the newly created group membership in the Identity Store.
+
+## Import
+
+`aws_identitystore_group_membership` can be imported using the `identity_store_id/membership_id`, e.g.,
+
+```
+$ terraform import aws_identitystore_group.example d-0000000000/00000000-0000-0000-0000-000000000000
```
Comment (bschaatsbergen) [inline website/docs/r/identitystore_group_membership.html.markdown:64]:
```suggestion
$ terraform import aws_identitystore_group_membership.example d-0000000000/00000000-0000-0000-0000-000000000000
```

File: website/docs/r/identitystore_group_membership.html.markdown [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+page_title: "AWS: aws_identitystore_group_membership"
+description: |-
+Terraform resource for managing an AWS IdentityStore Group Membership.
+---
+
+# Resource: aws_identitystore_group_membership
+
+Terraform resource for managing an AWS IdentityStore Group Membership.
+
+## Example Usage
```
Comment (bschaatsbergen) [inline website/docs/r/identitystore_group_membership.html.markdown:13]: Could you rename the `test` to `example` to be in line with the other identitystore docs?

File: website/docs/r/identitystore_group_membership.html.markdown [hunk 3]
```diff
@@ -0,0 +1,65 @@
+---
+subcategory: "IdentityStore"
```
Comment (bschaatsbergen) [inline website/docs/r/identitystore_group_membership.html.markdown:2]:
```suggestion
subcategory: "SSO Identity Store"
```

File: internal/service/identitystore/group_membership_test.go [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+			return create.Error(names.IdentityStore, create.ErrActionCheckingExistence, tfidentitystore.ResNameGroupMembership, rs.Primary.ID, err)
+		}
+
+		*groupMembership = *resp
+
+		return nil
+	}
+}
+
+func testAccPreCheck(t *testing.T) {
```
Comment (bschaatsbergen) [inline internal/service/identitystore/group_membership_test.go:234]: I believe in main there's already a testAccPreCheck function, could you rebase on main and remove this duplicate code?

**Review by bschaatsbergen — COMMENTED** [review]
few drive by comments; nice work so far.

**Comment by BenGisc** [comment]
Thanks Bruno, applied your suggestions ! (do not pay attention to the base branch change, was a misconfig on my side)

File: internal/service/identitystore/group_membership_test.go (hunk 4 above)
Comment (BenGisc) [reply internal/service/identitystore/group_membership_test.go]: Done

**Comment by BenGisc** [comment]
Seems like the rebase applied a lot of labels on my PR. Since it is my first contribution, and I would like to contribute more in the future, can you tell me what I did wrong please ?

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccIdentityStoreGroupMembership_' PKG=identitystore ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/identitystore/... -v -count 1 -parallel 2  -run=TestAccIdentityStoreGroupMembership_ -timeout 180m
=== RUN   TestAccIdentityStoreGroupMembership_basic
=== PAUSE TestAccIdentityStoreGroupMembership_basic
=== RUN   TestAccIdentityStoreGroupMembership_disappears
=== PAUSE TestAccIdentityStoreGroupMembership_disappears
=== RUN   TestAccIdentityStoreGroupMembership_GroupId
=== PAUSE TestAccIdentityStoreGroupMembership_GroupId
=== RUN   TestAccIdentityStoreGroupMembership_MemberId
=== PAUSE TestAccIdentityStoreGroupMembership_MemberId
=== CONT  TestAccIdentityStoreGroupMembership_basic
=== CONT  TestAccIdentityStoreGroupMembership_GroupId
--- PASS: TestAccIdentityStoreGroupMembership_basic (22.57s)
=== CONT  TestAccIdentityStoreGroupMembership_MemberId
--- PASS: TestAccIdentityStoreGroupMembership_GroupId (38.89s)
=== CONT  TestAccIdentityStoreGroupMembership_disappears
--- PASS: TestAccIdentityStoreGroupMembership_disappears (17.15s)
--- PASS: TestAccIdentityStoreGroupMembership_MemberId (36.39s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/identitystore	62.995s
```

**Comment by ewbankkit** [comment]
@BenGisc Thanks for the contribution :tada: :clap:.

There must have been some glitch with the PR labeler GitHub Action that caused all those labels to be added.
I manually tidied them up.

**Pull request merged**